### PR TITLE
Optimize VolumeRestriction plugin

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**

 /kind feature

**What this PR does / why we need it**:
Avoids redundant iterations in the VolumeRestrictions scheduler plugin. This may have positive impact for workloads that use secrets/configmaps etc.

**Which issue(s) this PR fixes**:
Part of #93217

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @wojtek-t

### CODE REVIEW

**Comment by ahg-g** [comment]
unlikely to cause too much difference, but worth doing it.

File: pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go [hunk 1]
```diff
@@ -118,7 +113,13 @@ func haveOverlap(a1, a2 []string) bool {
 // - Ceph RBD forbids if any two pods share at least same monitor, and match pool and image, and the image is read-only
 // - ISCSI forbids if any two pods share at least same IQN and ISCSI volume is read-only
 func (pl *VolumeRestrictions) Filter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
-	for _, v := range pod.Spec.Volumes {
+	for i := range pod.Spec.Volumes {
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:116]: It's no harm to keep the original `for _, v := range pod.Spec.Volumes` logic, no?

**Review by Huang-Wei — COMMENTED** [review]
One question. LGTM otherwise.

File: pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:116]: why should we copy the object twice?

Comment (Huang-Wei) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:116]: True, but I'd argue the cost (the size of `Volume` is trivial) is trivial.

/lgtm
anyway

**Comment by wojtek-t** [comment]
LGTM - thanks!

**Review by hasheddan — APPROVED** [review]
/lgtm
/kind feature

**Pull request merged**